autoload/neomake/makers/clippy.vim

```diff
@@ -0,0 +1,29 @@
+" vim: ts=4 sw=4 et
+
+function! neomake#makers#clippy#clippy() abort
+    let errorfmt = '%Eerror[E%n]: %m,'.
+                 \ '%Eerror: %m,'.
+                 \ '%Wwarning: %m,'.
+                 \ '%Inote: %m,'.
+                 \ '%-Z\ %#-->\ %f:%l:%c,'.
+                 \ '%I\ %#\= %t%*[^:]: %m,'.
+                 \ '%I\ %#|\ %#%\\^%\\+ %m,'.
+                 \ '%-G%s,'
+
+    " When rustup and a nightly toolchain is installed, that is used.
+    " Otherwise, the default cargo exectuable is used. If this is not part
+    " of a nightly rust, this will fail.
+    if executable('rustup') && system('rustup show | grep nightly | wc -l') >= 1
```

Comment:
I'll look into the caching.
rustup manages multiple installed versions of the rust toolchain, `--version` would only tell the version of rustup, which is not of interest hier. `rustup show` shows the installed toolchain versions, for example:
```
Default host: unknown-linux-gnu

installed toolchains
--------------------

stable-x86_64-unknown-linux-gnu (default)
nightly-x86_64-unknown-linux-gnu

active toolchain
----------------

stable-x86_64-unknown-linux-gnu (default)
rustc 1.11.0 (9b21dcd6a 2016-08-15)
```

It might well be that there is a more efficient way to do this, but I don't know it.